lib/ansible/modules/cloud/amazon/code_build.py

```diff
@@ -0,0 +1,338 @@
+          description: The build spec declaration to use for the builds in this build project.
+          returned: always
+          type: string
+        auth:
+          desription: Information about the authorization settings for AWS CodeBuild to access the source code to be built.
+          returned: when configured
+          type: complex
+        insecure_ssl:
+          description: True if set to ignore SSL warnings.
+          returned: when configured
+          type: bool
+    artifacts:
+      description: Information about the output of build artifacts
+      returned: always
+      type: complex
+      contains:
+        type:
+          description: The type of build artifact.
+          returned: always
+          type: string
+          sample: CODEPIPELINE
+        location:
+          description: Output location for build artifacts
+          returned: when configured
+          type: string
+        # and more... see http://boto3.readthedocs.io/en/latest/reference/services/codebuild.html#CodeBuild.Client.create_project
+    cache:
+      description: Cache settings for the build project.
+      returned: when configured
+      type: complex
+    environment:
+      description: Environment settings for the build
+      returned: always
+      type: complex
+    service_role:
+      description: IAM role to be used during build to access other AWS services.
+      returned: always
+      type: string
+      sample: arn:aws:iam::123123123:role/codebuild-service-role
+    timeout_in_minutes:
+      description: The timeout of a build in minutes
+      returned: always
+      type: int
+      sample: 60
+    tags:
+      description: Tags added to the project
+      returned: when configured
+      type: list
+    created:
+      description: Timestamp of the create time of the project
+      returned: always
+      type: string
+      sample: 2018-04-17T16:56:03.245000+02:00
+    last_modified:
+      description: Timestamp of the time the project was last updated.
+      returned: always
+      type: string
+      sample: 2018-04-17T16:56:03.245000+02:00
+'''
+
+import traceback
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict, snake_dict_to_camel_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def create_or_update_project(client, params, module):
+    resp = {}
+    name = params['name']
+    # Sanity check and cleanup params not needed for boto methods:
+    if not isinstance(name, str):
+        module.fail_json(msg="Params was missing name", exception=traceback.format_exc())
+    clean_params = dict((k, v) for k, v in params.items() if v is not None)
+    clean_params.pop('region', None)
+    clean_params.pop('state', None)
+    clean_params.pop('validate_certs', None)
```

Comment:
Since there could always be incompatible parameters added to ec2_argument_spec that would then need to be popped here, I think rather than removing parameters you should just make a fresh dict and add the ones accepted by this module.